lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/RunningWorkers.java

```diff
@@ -132,6 +138,13 @@ void launchTasklet(final Tasklet tasklet) {
           }
         }
 
+        if (cancelledTasklets.contains(tasklet.getId())) {
+          tasklet.cancelled();
+          cancelledTasklets.remove(tasklet.getId());
```

Comment:
Removing tasklet id from the set immediately would not work if we enable duplicate scheduling. It would not be addressed in this PR, but I think it'd not be bad to leave a short comment about it.